src/components/Tooltip/Tooltip.tsx

```diff
@@ -56,113 +59,268 @@ const paddingsBottom = {
+    ["bottom-End"]: "bottom-end",
+    ["left-Start"]: "left-start",
+    ["left-End"]: "left-end",
+    ["right-Start"]: "right-start",
+    ["right-End"]: "right-end",
+};
+
+export const Tooltip: FC<TooltipProps> = ({
+    content,
+    tooltipIcon,
+    heading,
+    headingIcon,
+    linkUrl,
+    linkLabel,
+    brightHeader,
+    buttons,
+    children,
+    position = TooltipPosition.Bottom,
+    alignment = TooltipAlignment.Middle,
+    withArrow,
+    flip = true,
+    triggerRefElement,
+    hoverDelay = 200,
+}) => {
     const linkRef = useRef<HTMLAnchorElement | null>(null);
     const { linkProps } = useLink({}, linkRef);
-
     const { isFocusVisible, focusProps } = useFocusRing();
+    const hasLargePaddingTop = linkUrl || brightHeader || buttons || heading || headingIcon;
 
-    const { tooltipProps } = useTooltip(tooltipAriaProps);
+    const placement = alignment === "Middle" ? position : placementMap[`${position}-${alignment}`];
+    const tooltipContainerRef = useRef<HTMLDivElement | null>(null);
+    const [arrowElement, setArrowElement] = useState<HTMLDivElement | null>(null);
 
-    const hasLargePaddingTop = linkUrl || brightHeader || buttons || heading || headingIcon;
+    const setArrowClasses = () => {
+        const currentPlacement = popperInstance.state?.placement ?? position;
+        switch (true) {
+            case currentPlacement.toString().includes(TooltipPosition.Top):
+                return "before:tw-border-t-0 before:tw-border-l-0 tw-bottom-[-6px] before:tw-bg-black-100 before:dark:tw-bg-white";
+            case currentPlacement.toString().includes(TooltipPosition.Right):
+                return merge([
+                    "before:tw-border-t-0 before:tw-border-r-0 tw-left-[-6px]",
+                    brightHeader && alignment === TooltipAlignment.Start
+                        ? brightHeaderArrowBackgroundColors[brightHeader]
+                        : "before:tw-bg-black-100 before:dark:tw-bg-white",
+                ]);
+            case currentPlacement.toString().includes(TooltipPosition.Bottom):
+                return merge([
+                    "before:tw-border-b-0 before:tw-border-r-0 tw-top-[-6px]",
+                    brightHeader
+                        ? brightHeaderArrowBackgroundColors[brightHeader]
+                        : "before:tw-bg-black-100 before:dark:tw-bg-white",
+                ]);
+            case currentPlacement.toString().includes(TooltipPosition.Left):
+                return merge([
+                    "before:tw-border-b-0 before:tw-border-l-0 tw-right-[-6px]",
+                    brightHeader && alignment === TooltipAlignment.Start
+                        ? brightHeaderArrowBackgroundColors[brightHeader]
+                        : "before:tw-bg-black-100 before:dark:tw-bg-white",
+                ]);
+            default:
+                return "before:tw-border-b-0 before:tw-border-r-0 tw-top-[-6px] before:tw-bg-black-100 before:dark:tw-bg-white";
```

Comment:
Can we make it as a `const` outside the component?
If not, we can at least wrap it in a `useCallback`